thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -140,21 +140,49 @@ class CDistributorBase : implements IHashDistributor, implements IExceptionHandl
              */
             dedupList.sort(*iCompare, 1);
 
-            OwnedConstThorRow prev;
-            for (unsigned i = c; i>0;)
+            if (!keepBestCompare)
             {
-                OwnedConstThorRow row = dedupList.getClear(--i);
-                if ((NULL != prev.get()) && (0 == iCompare->docompare(prev, row)))
+                OwnedConstThorRow prev;
+                for (unsigned i = c; i>0;)
                 {
-                    /* NB: do not alter 'total' size. It represents the amount originally added to the bucket
-                     * which will be deducted when sent.
-                     */
+                    OwnedConstThorRow row = dedupList.getClear(--i);
+                    if ((NULL != prev.get()) && (0 == iCompare->docompare(prev, row)))
+                    {
+                        /* NB: do not alter 'total' size. It represents the amount originally added to the bucket
+                        * which will be deducted when sent.
+                        */
+                    }
+                    else
+                    {
+                        prev.set(row);
+                        rows.enqueue(row.getClear());
```

Comment:
Not sure if my comment was always on wrong line, or if it has moved during squashing or something. But it was supposed to be associated with the very similar lines in the else block of if (!keepBestCompare).

Anyway, can you comment on the scenario above, in the else block ?